### PR TITLE
Remove instant execution workarounds for AGP 3.x

### PR DESCRIPTION
**Pull request by eskatos**
As they are not needed anymore given instant execution now targets 4.x.

### CODE REVIEW

File: subprojects/instant-execution/src/main/kotlin/org/gradle/instantexecution/serialization/Workarounds.kt [hunk 1]
```diff
@@ -18,27 +18,19 @@ package org.gradle.instantexecution.serialization
...
 
     private
-    val ignoredBeanFields = listOf(
-        // Ignore a lambda field for now
-        "mFolderFilter" to "com.android.ide.common.resources.DataSet"
-    )
+    val ignoredBeanFields: List<Pair<String, String>> = emptyList()
 
     fun isIgnoredBeanField(field: Field) =
         ignoredBeanFields.contains(field.name to field.declaringClass.name)
```
Comment (adammurdoch) [inline subprojects/instant-execution/src/main/kotlin/org/gradle/instantexecution/serialization/Workarounds.kt]: shouldn't this just be `= false`? which means we can remove the method.

File: subprojects/instant-execution/src/main/kotlin/org/gradle/instantexecution/serialization/Workarounds.kt [hunk 2]
```diff
@@ -18,27 +18,19 @@ package org.gradle.instantexecution.serialization
...
     private
-    val staticFieldsByTypeName = mapOf(
-        "com.android.build.gradle.internal.tasks.Workers" to mapOf(
-            "aapt2ThreadPool" to { ForkJoinPool(8) }
-        )
-    )
+    val staticFieldsByTypeName: Map<String, Map<String, () -> Any?>> = emptyMap()
 
     fun maybeSetDefaultStaticStateIn(scope: ClassLoaderScope) {
         listOf(scope.localClassLoader, scope.exportClassLoader).forEach { loader ->
```
Comment (adammurdoch) [inline subprojects/instant-execution/src/main/kotlin/org/gradle/instantexecution/serialization/Workarounds.kt]: This method also does not do anything any more and can be removed.

File: subprojects/instant-execution/src/main/kotlin/org/gradle/instantexecution/serialization/Workarounds.kt (hunk 1 above)
Comment (eskatos) [reply subprojects/instant-execution/src/main/kotlin/org/gradle/instantexecution/serialization/Workarounds.kt]: Yep, I'll remove `Workarounds` altogether.

File: subprojects/instant-execution/src/main/kotlin/org/gradle/instantexecution/serialization/Workarounds.kt (hunk 2 above)
Comment (eskatos) [reply subprojects/instant-execution/src/main/kotlin/org/gradle/instantexecution/serialization/Workarounds.kt]: Yep, I'll remove `Workarounds` altogether.

**Pull request merged**